.travis.yml

```diff
@@ -1,15 +1,53 @@
 language: rust
-rust:
-  - nightly
-  - beta
-  - stable
-env:
-  - GTK=3.4
-  - GTK=3.18
+matrix:
+  include:
+    - os: linux
+      rust: nightly
+      env: GTK=3.4
+    - os: linux
+      rust: nightly
+      env: GTK=3.18
+    - os: linux
+      rust: beta
+      env: GTK=3.4
+    - os: linux
+      rust: beta
+      env: GTK=3.18
+    - os: linux
+      rust: stable
+      env: GTK=3.4
+    - os: linux
+      rust: stable
+      env: GTK=3.18
+    - os: osx
+      rust: nightly
+      env: GTK=3.4
+    # - os: osx
+    #   rust: nightly
+    #   env: GTK=3.18
+    - os: osx
+      rust: beta
+      env: GTK=3.4
+    # - os: osx
+    #   rust: beta
+    #   env: GTK=3.18
+    - os: osx
+      rust: stable
+      env: GTK=3.4
+    # - os: osx
+    #   rust: stable
+    #   env: GTK=3.18
+sudo: true
 addons:
   apt:
     packages:
     - libgtk-3-dev
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew unlink python; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install gtk+3 cairo atk; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/opt/X\
+11/lib/pkgconfig; fi
```

Comment:
Maybe it be clearer is use multiline?
```
  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
    export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/opt/X11/lib/pkgconfig; 
    fi
```

Reply:
If you prefer. Don't really have any preference in here...